### DISCUSSION TITLE
Include button in SelectItem, ComboboxItem

### DISCUSSION THREAD

**Discussion by mule-stand** [ariakit/ariakit#3765] [Questions]
Hello! Can I include button directly in SelectItem, ComboboxItem (like "Delete Item", etc)? And if dont, how I can do it? And how i can show this button only when item is active (like data-[active-item])?

Like this 
![изображение](https://github.com/ariakit/ariakit/assets/113935919/863c49e8-a3fa-4d2a-a01c-8a06515c3c31)

**Comment by diegohaz** [comment]
That's a tough one since the current ARIA spec doesn't provide a way to represent this type of UI. There's a proposed `aria-actions` attribute that could potentially address this in the future: https://github.com/w3c/aria/issues/1440

I plan to write an example of this, but I need to do more research first. My top idea at the moment is to render those action elements outside of the combobox/select list, and then position them next to their respective items using CSS/Floating UI.

You can render these elements conditionally by checking if the combobox/select [`activeId`](https://ariakit.org/reference/use-combobox-store#activeid) state corresponds to the current item:

```tsx
function MyComboboxItem(props) {
  const id = React.useId();
  const combobox = Ariakit.useComboboxContext()!;
  const isActiveItem = combobox.useState((state) => state.activeId === id);
```

**Reply by mule-stand** [reply]
useId is just an example of the concept? Do I need to specify an id for each SelectItem? Or how? Because I understand there are some identifiers generated under the hood?

**Reply by diegohaz** [reply]
Ariakit automatically generates IDs under the hood, but if you want to use them in your code, it's best to generate them yourself using React's `useId`.
